packages/query/src/index.ts

```diff
@@ -558,14 +561,14 @@ const getQueryOptionsDefinition = ({
         : `typeof ${operationName}`
     }>>`;
 
-    return `${prefix}${pascal(type)}Options<${funcReturnType}, TError, TData${
+    return `${partialOptions ? 'WithOptional<':''}${prefix}${pascal(type)}Options<${funcReturnType}, TError, TData${
```

Comment:
Why add something new and not just use Partial?

Reply:
My thought here was to just target the one troublesome property only and let any similar future changes get caught early rather than fail at query time. If react-query made a similar change in the future, typescript checking would error the same as `queryKey` did here, rather than allow a required property be passed undefined and cause errors down the pipeline.